cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/common/RandomCluster.java

```diff
@@ -128,14 +128,13 @@ public static void populate(ClusterModel cluster,
     }
     // Increase the replication factor.
     for (int i = 0; i < properties.get(ClusterProperty.NUM_TOPICS).intValue(); i++) {
-      int oldReplicationFactor = metadata.get(i).replicationFactor();
       int randomReplicationFactor = uniformlyRandom(properties.get(ClusterProperty.MIN_REPLICATION).intValue(),
           properties.get(ClusterProperty.MAX_REPLICATION).intValue(), TestConstants.REPLICATION_SEED + i);
       metadata.get(i).setReplicationFactor(randomReplicationFactor);
 
       if (totalTopicReplicas(metadata) > properties.get(ClusterProperty.NUM_REPLICAS).intValue()) {
         // Rollback to previous replicationFactor.
-        metadata.get(i).setReplicationFactor(oldReplicationFactor);
+        metadata.get(i).setReplicationFactor(properties.get(ClusterProperty.MIN_REPLICATION).intValue());
```

Comment:
Just curious why do we need this change?

Reply:
To prevent the replication factor value from being set to a value under `MIN_REPLICATION` -- i.e. to the default value of metadata.